Generator produces invalid code for plugins with parameterized Components

## Description

The `labelle generate` command produces code in `main.zig` that doesn't compile when using plugins like `labelle-tasks` and `labelle-pathfinding`.

## Steps to Reproduce

1. Add plugins to `project.labelle`:
```zig
.plugins = .{
    .{ .name = "labelle-pathfinding", .version = "2.5.0" },
    .{ .name = "labelle-tasks", .version = "0.3.0" },
},
```

2. Run `labelle generate`

3. Run `zig build`

## Expected Behavior

The generated code should compile successfully.

## Actual Behavior

Build fails with errors:

```
main.zig:84:24: error: root source file struct 'pathfinding' has no member named 'Components'
    labelle_pathfinding.Components,
```

## Root Cause

The generator produces this code in `main.zig`:

```zig
pub const Components = engine.ComponentRegistryMulti(.{
    struct {
        // ... local components ...
    },
    labelle_pathfinding.Components,
    labelle_tasks.Components,
});
```

However:
1. **labelle-pathfinding v2.5.0** doesn't export a `Components` type at its root
2. **labelle-tasks v0.3.0** exports `Components` as a generic function `Components(ItemType)`, not a type

## Suggested Fix

The generator should:
1. Only include `plugin.Components` if the plugin actually exports it
2. Handle parameterized Components (like `labelle-tasks.Components(ItemType)`) - perhaps via a plugin manifest that specifies the item type, or skip auto-registration for such plugins

## Environment

- Zig: 0.15.1
- labelle-engine: v0.8.0
- labelle-pathfinding: v2.5.0  
- labelle-tasks: v0.3.0

🤖 Generated with [Claude Code](https://claude.ai/code)

--- a/src/generator.zig
+++ b/src/generator.zig
@@ -188,7 +188,14 @@ fn generateMainZigRaylib(
     var buf: std.ArrayListUnmanaged(u8) = .{};
     const writer = buf.writer(allocator);
 
-    const has_plugins = config.plugins.len > 0;
+    // Check if any plugin contributes Components
+    var has_plugin_components = false;
+    for (config.plugins) |plugin| {
+        if (plugin.components != null) {
+            has_plugin_components = true;
+            break;
+        }
+    }
 
     // Pre-compute sanitized plugin names for Zig identifiers
     var plugin_zig_names = try allocator.alloc([]const u8, config.plugins.len);
@@ -243,10 +250,11 @@ fn generateMainZigRaylib(
         try zts.print(main_raylib_tmpl, "prefab_registry_end", .{}, writer);
     }
 
-    // Component registry - use ComponentRegistryMulti when plugins are present
-    if (has_plugins) {
+    // Component registry - use ComponentRegistryMulti when plugins contribute Components
+    if (has_plugin_components) {
         // Use ComponentRegistryMulti to merge base components with plugin components
         if (components.len == 0) {
+            // Empty start template already includes closing brace
             try zts.print(main_raylib_tmpl, "component_registry_multi_empty_start", .{}, writer);
         } else {
             try zts.print(main_raylib_tmpl, "component_registry_multi_start", .{}, writer);
@@ -254,11 +262,13 @@ fn generateMainZigRaylib(
                 const type_name = capitalize(name);
                 try zts.print(main_raylib_tmpl, "component_registry_multi_item", .{ type_name[0..name.len], type_name[0..name.len] }, writer);
             }
+            try zts.print(main_raylib_tmpl, "component_registry_multi_base_end", .{}, writer);
         }
-        try zts.print(main_raylib_tmpl, "component_registry_multi_base_end", .{}, writer);
-        // Add plugin Components
-        for (plugin_zig_names) |zig_name| {
-            try zts.print(main_raylib_tmpl, "component_registry_multi_plugin", .{zig_name}, writer);
+        // Add plugin Components (only for plugins that specify a components field)
+        for (config.plugins, 0..) |plugin, i| {
+            if (plugin.components) |components_expr| {
+                try zts.print(main_raylib_tmpl, "component_registry_multi_plugin", .{ plugin_zig_names[i], components_expr }, writer);
+            }
         }
         try zts.print(main_raylib_tmpl, "component_registry_multi_end", .{}, writer);
     } else {

--- a/src/project_config.zig
+++ b/src/project_config.zig
@@ -57,6 +57,12 @@ pub const Plugin = struct {
     /// Module name exported by the package (e.g., "pathfinding" for labelle-pathfinding)
     /// If not provided, defaults to the plugin name with hyphens replaced by underscores
     module: ?[]const u8 = null,
+    /// Components type exported by the plugin. If null, no Components are included.
+    /// Examples:
+    ///   - null: don't include any Components from this plugin (default)
+    ///   - "Components": use plugin.Components
+    ///   - "Components(MyItem)": use plugin.Components(MyItem) for parameterized types
+    components: ?[]const u8 = null,
 };
 
 /// Atlas resource declaration

--- a/src/templates/main_raylib.txt
+++ b/src/templates/main_raylib.txt
@@ -80,7 +80,7 @@ pub const Components = engine.ComponentRegistryMulti(.{{
 .component_registry_multi_base_end
     }},
 .component_registry_multi_plugin
-    {s}.Components,
+    {s}.{s},
 .component_registry_multi_end
 }});
 .script_registry_empty

